mocks/uid.go

```diff
@@ -4,13 +4,16 @@ import (
 	"github.com/pborman/uuid"
 )
 
-// uuid.UUID is a slice and cannot be used as a map key.
+// UID is defined since uuid.UUID is a slice and cannot be used as a map key.
 type UID [16]byte
 
+// Interface converts the fixed length array representation to the uuid package
+// type.
 func (u *UID) Interface() uuid.UUID {
 	return uuid.UUID(u[:])
 }
 
+// IsNIL returns true if the UID value is 0.
 func (u *UID) IsNIL() bool {
 	for i := 0; i < 16; i++ {
 		if i != 0 {
```

Comment:
then I think you should do u[i] != 0
or
more simply -> https://play.golang.org/p/7LA21ov9g6